### ISSUE TITLE
Custom Model 

### ISSUE THREAD

**Issue by jim-carter-12** [MaximIntegratedAI/ai8x-training#178]
Hi everyone,

I am working on keyword detection. So, I trained model like your repository. I created the test audio files. After Model trained, I need to see accuracy. Thats why I should make Confision Matrix. But I trained model, I saved like that;
`torch.save(model.state_dict(), PATH)`

I couldn't load like that;
`model.load_state_dict(torch.load(PATH))`

Would help me please? After solving this problem. I want to know How I load test dataset into the custom model?

**Comment by aniktash** [comment]
Hi,
You can use our train.py pipeline to train your model and it will take care of saving checkpoints and loading them, and loading datasets. To save and load checkpoints, it uses distiller/apputils/save_checkpoint() and distiller/apputils/load_lean_checkpoint() functions. 
If checkpoint is correct, you may use  torch.load as well. See the following code snippet:

# input new checkpoint and model
trained_checkpoint_path = "ai85-kws20_v3-qat8-q.pth.tar"  # from ai8x-synthesis/trained
checkpoint = torch.load(trained_checkpoint_path)
mod = importlib.import_module("ai85net-kws20-v3")
model = mod.AI85KWS20Netv3(num_classes=21, num_channels=128, dimensions=(128, 1), bias=False, quantize_activation=True, weight_bits=8, bias_bits=8)
ai8x.fuse_bn_layers(model)
ai8x.initiate_qat(model, qat_policy)
model.load_state_dict(checkpoint['state_dict'], strict=False)
ai8x.update_model(model)

You can check the dataset/kws20.py to see how the dataset is created and loaded. There is also appnote on dataloader:
https://www.maximintegrated.com/en/design/technical-documents/app-notes/7/7600.html

**Comment by oussemajelassi** [comment]
Hi how can I when using train.py specify my dataset ? please note that i am using a whole different model than the implemented  examples ! In general I am facing some issues understanding the procedure of training a different model.
